Code Navigation IA revamp (#202)
sourcegraph/docs: docs/code-search/code-navigation/precise_code_navigation.mdx

# Precise Code Navigation

<p className="subtitle">Learn and understand about Precise Code Navigation.</p>

Precise Code Navigation is an opt-in feature that is enabled from your admin settings and requires you to upload indexes for each repository to your Sourcegraph instance. Once setup is complete on Sourcegraph, precise code navigation is available for use across popular development tools:

- On the Sourcegraph web UI
- On code files on your code host, via [integrations](/integration/)
- On diffs in your code review tool, via integrations
- Via the [Sourcegraph API](/api/graphql)

Sourcegraph automatically uses Precise Code Navigation whenever available, and Search-based Code Navigation is used as a fallback when precise navigation is not available.

> Precise code navigation relies on the following technologies:
> 
> - [SCIP](https://github.com/sourcegraph/scip) (SCIP Code Intelligence Protocol) and
> - [LSIF](https://github.com/Microsoft/language-server-protocol/blob/master/indexFormat/specification)
> (Language Server Index Format) data to deliver precomputed code navigation. It provides fast and highly accurate code navigation but needs to be periodically generated and uploaded to your Sourcegraph instance. Precise code navigation is an opt-in feature: repositories for which you have not uploaded indexes will continue to use the search-based code navigation.

Reviewer: ```suggestion
Precise code navigation relies on the open source [SCIP Code Intelligence Protocol](https://github.com/sourcegraph/scip), which is a language-agnostic protocol for indexing source code.
```

We can remove the reference to LSIF as we no longer support LSIF directly, we only support it via conversion to SCIP.

> # Precise-based Code Navigation

Reviewer: ```suggestion
# Precise Code Navigation
```

4. Set up **auto-dependency indexing** to navigate and search through the dependencies your code uses:
    - **Go**: Enable [auto-indexing](/code_navigation/explanations/auto_indexing) and Sourcegraph will start indexing your dependencies.
    - **JavaScript, TypeScript**: Enable [auto-indexing](/code_navigation/explanations/auto_indexing) and set up an [npm dependencies code host](/integration/npm).
    - **Java, Scala, Kotlin**: Enable [auto-indexing](/code_navigation/explanations/auto_indexing) and set up a [JVM dependencies code host](/integration/jvm).

## Supported languages and indexers

Precise Code Navigation requires language-specific indexes to be generated and uploaded to your Sourcegraph instance. We currently have precise code navigation support for the languages below. See the [indexers page](/code_navigation/references/indexers) for a detailed breakdown of each indexer's status.

|        Language        |                                      Indexer                                      |        Status         |
| ---------------------- | --------------------------------------------------------------------------------- | --------------------- |

> | Go                     | [lsif-go](https://sourcegraph.com/github.com/sourcegraph/lsif-go)                 | 🟢 Generally available |

Reviewer 2: This should be scip-go, I think I updated that in main but since the file is moved/new it didnt map
